edweiss-app/components/quiz/LectureQuizComponents.tsx

```diff
@@ -0,0 +1,162 @@
+	// 	const res = await callFunction(Quizzes.Functions.toggleLectureQuizResult, { lectureId: lectureId, lectureEventId: lectureEventId, courseId: courseId, });
+
+	// 	if (res.status === 1) {
+	// 		console.log(`toggled showResultToStudent boolean`);
+	// 	} else {
+	// 		console.log(`Error while toggling boolean shoresultToStudent`);
+	// 	}
+	// 	setLoading(false);
+	// }
+
+	// --------------------------
+	if (quizEvent == undefined || studentAttempts == undefined) {
+		return <TActivityIndicator testID='undefined-quiz-loading-prof' />;
+	}
+
+	const studentAttemptsData = studentAttempts.map(doc => doc.data);
+
+
+	return (
+		<>
+			<TSafeArea>
+				{quiz?.showResultToStudents && studentAttempts.length > 0 && <>
+					<TView mb='lg'>
+						<SingleDistributionDisplay exercise={quiz.exercise} exerciseAttempts={studentAttemptsData} />
+					</TView>
+				</>
+				}
+				{quizEvent != undefined && !quiz?.showResultToStudents && <>
+					<TView>
+						<TText> Quiz is live! </TText>
+					</TView>
+				</>
+				}
+				{/* ------- must be put in the remote control ------- */}
+				{/* <FancyButton loading={loading} onPress={toggleResult}>
+					{quiz?.showResultToStudents ? "Show quiz to students" : "Show results to students"}
+				</FancyButton> */}
+			</TSafeArea>
+
+		</>
+	);
+
+};
+
+export const LectureQuizStudentView: ReactComponent<{ courseId: string, lectureId: string, lectureEventId: string, pathToLectureEvents: string, pathToAttempts: string, quizEvent: Document<LectureDisplay.QuizLectureEvent> | undefined }> = ({ courseId, lectureId, lectureEventId, pathToLectureEvents, pathToAttempts, quizEvent }) => {
+
+	const { uid } = useAuth();
+	const previousAttempt = useDoc(CollectionOf<LectureQuizzesAttempts.LectureQuizAttempt>(pathToAttempts), uid);
+	const [studentAnswer, setStudentAnswer] = useState<QuizzesAttempts.Answer | undefined>(undefined);
+	const quiz = quizEvent?.data.quizModel
+	const exercise = quiz?.exercise;
+
+
+	useEffect(() => {
+
+		if (quiz == undefined || exercise == undefined)
+			return;
+
+		const defaultAnswer = () => {
+			if (exercise.type == "MCQ") {
+				const MCQAnswer: QuizzesAttempts.MCQAnswersIndices = { type: "MCQAnswersIndices", value: [] }
+				return MCQAnswer;
+			}
+			else {
+				const TFAnswer: QuizzesAttempts.TFAnswer = { type: "TFAnswer", value: undefined }
+				return TFAnswer;
+			}
+		}
+
+		setStudentAnswer(defaultAnswer());
+	}, [quizEvent]);
+
+
+
+	const onUpdate = useCallback((newAnswer: number[] | boolean | undefined) => {
+		const newFormattedAnswer: LectureQuizzesAttempts.LectureQuizAttempt = exercise?.type == "MCQ" ? { type: "MCQAnswersIndices", value: newAnswer as number[] } : { type: "TFAnswer", value: newAnswer as boolean }
+		setStudentAnswer(newFormattedAnswer)
+	}, [exercise]);
+
+	if (quizEvent == undefined || quiz == undefined || exercise == undefined) {
+		return (<TActivityIndicator testID='undefined-quizEvent-loading' />)
+	}
+
+	if (typeof lectureId != 'string') {
```

Comment:
Why this check? `lectureId` is necessarily a string

Reply:
It is indeed unnecessary.